### PR TITLE
Fix various lgtm warning

### PR DESCRIPTION
**Pull request by 08A**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
Start solving lgtm's warning. Some file don't respect coding style, so there is a big diff.

**Test plan**
Rerun lgtm

**Work on issues**
 #16493

### CODE REVIEW

File: libr/bin/format/mach0/mach0.c [hunk 1]
```diff
@@ -2826,7 +2826,8 @@ const struct symbol_t *MACH0_(get_symbols)(struct MACH0_(obj_t) *bin) {
 }
 
 static int parse_import_ptr(struct MACH0_(obj_t) *bin, struct reloc_t *reloc, int idx) {
-	int i, j, sym, wordsize;
+	int i, j;
+	size_t sym, wordsize;
```
Comment (ret2libc) [inline libr/bin/format/mach0/mach0.c:2830]: Is this right? It seems to me the value of `j` is assigned to `sym` few lines below, so they should be of the same time. Also, `sym` initially has -1 as value, so size_t is probably not the right type.

File: libr/core/cmd_print.c [hunk 2]
```diff
@@ -4671,7 +4671,7 @@ static int cmd_print(void *data, const char *input) {
 	ut32 tbs = core->blocksize;
 	ut64 n, off, from, to, at, ate, piece;
 	ut64 tmpseek = UT64_MAX;
-	const int addrbytes = core->io->addrbytes;
+	const size_t addrbytes = core->io->addrbytes;
```
Comment (ret2libc) [inline libr/core/cmd_print.c:4674]: `addrbytes` is `int` in `RIO`, why this change?

File: libr/asm/asm.c [hunk 3]
```diff
@@ -617,8 +617,8 @@ R_API RAsmCode* r_asm_mdisassemble(RAsm *a, const ut8 *buf, int len) {
 	ut64 pc = a->pc;
 	RAsmOp op;
 	ut64 idx;
-	int ret, slen;
-	const int addrbytes = a->user ? ((RCore *)a->user)->io->addrbytes : 1;
+	size_t ret, slen;
+	const size_t addrbytes = a->user? ((RCore *)a->user)->io->addrbytes: 1;
```
Comment (ret2libc) [inline libr/asm/asm.c:621]: addrbytes in RIO is int, why this change?

File: libr/core/visual.c [hunk 4]
```diff
@@ -254,7 +254,7 @@ static bool __core_visual_gogo (RCore *core, int ch) {
 				break;
 			}
 			(void)p->consbind.get_size (&scr_rows);
-			int scols = r_config_get_i (core->config, "hex.cols");
+			size_t scols = r_config_get_i (core->config, "hex.cols");
```
Comment (ret2libc) [inline libr/core/visual.c:257]: `r_config_get_i` returns a `ut64`, not `size_t`

File: libr/socket/run.c [hunk 5]
```diff
@@ -162,15 +162,16 @@ static char *getstr(const char *src) {
...
-				rep = atoi (src + 1);
+				rep = strtol (src + 1, NULL, 10);
 				len = strlen (pat);
 				if (rep > 0) {
-					char *buf = malloc (rep);
+					char *buf = malloc (rep * sizeof(char));
 					if (buf) {
-						for (i = 0; i < rep; i++) {
-							buf[i] = pat[i%len];
+						for (long i = 0; i < rep; i++) {
```
Comment (ret2libc) [inline libr/socket/run.c:173]: do not declar the var inside the for-loop. Apparently there are compilers which complain and we don't have the right flags in place to fix that at the moment.

File: libr/bin/format/mach0/mach0.c (hunk 1 above)
Comment (08A) [reply libr/bin/format/mach0/mach0.c:2830]: Indeed i will replace his type by ssize_t

File: libr/core/cmd_print.c (hunk 2 above)
Comment (08A) [reply libr/core/cmd_print.c:4674]: We have a possible overflow bellow, if the type is int.

```
size_t ... = int * int;
                 ^
                 | -- possible overflow
```

File: libr/asm/asm.c (hunk 3 above)
Comment (08A) [reply libr/asm/asm.c:621]: same than before

File: libr/core/visual.c (hunk 4 above)
Comment (08A) [reply libr/core/visual.c:257]: size_t >= ut64 i believe modern C should use size_t when there is a choice.

Comment (eagleoflqj) [reply libr/core/visual.c:257]: No size_t is 32-bit on i386. Try `gcc -m32 a.c` to print sizeof(size_t).

Comment (ret2libc) [reply libr/core/visual.c:257]: what @eagleoflqj says is true, moreover even just for consistency if a function returns `ut64` you should use the same type if possible.

Comment (08A) [reply libr/core/visual.c]: Okay, it is fixed

**Comment by 08A** [comment]
I was wandering why we use ut64, ut32 if there is stdint.h.

**Comment by ret2libc** [comment]
> I was wandering why we use ut64, ut32 if there is stdint.h.

@radare  ^

**Comment by XVilka** [comment]
Because stdint isn't (wasn't) on the every supported platform when we
started to use it* types. Radare2 supports A LOT of compilers and
platforms, keep this in mind.

File: libr/core/cmd_print.c (hunk 2 above)
Comment (ret2libc) [reply libr/core/cmd_print.c:4674]: I see, but I'm not sure that's the proper way to fix it (since you store the result in a `int` anyway). Maybe just check for overflows before doing the multiplication. It is probably wrong anyway if the number is so big.

Comment (eagleoflqj) [reply libr/core/cmd_print.c:4674]: What about a conversion like (size_t)int * int? Or change addrbytes to some ut if it doesn't use -1 to indicate error.

File: libr/asm/asm.c (hunk 3 above)
Comment (radare) [reply libr/asm/asm.c:621]: ? you should set RIO->addrbytes as size_t or whatever then

File: libr/bin/format/mach0/mach0.c [hunk 6]
```diff
@@ -2826,7 +2826,9 @@ const struct symbol_t *MACH0_(get_symbols)(struct MACH0_(obj_t) *bin) {
 }
 
 static int parse_import_ptr(struct MACH0_(obj_t) *bin, struct reloc_t *reloc, int idx) {
-	int i, j, sym, wordsize;
+	int i, j;
+	ssize_t sym;
```
Comment (radare) [inline libr/bin/format/mach0/mach0.c:2830]: sym is signed?

Comment (08A) [reply libr/bin/format/mach0/mach0.c:2830]: The default value is -1.

File: libr/core/cmd_print.c (hunk 2 above)
Comment (radare) [reply libr/core/cmd_print.c:4674]: that wont solve the problem. we have the UT64_ADD_OVFCHK macros that detect the overflow before it happens. but we have it for + not for *.  imho addrbytes shouldnt be larger than 4 or 8, so we probably dont need to be that paranoic on handling overflows in this operation

File: libr/core/cmd_print.c [hunk 7]
```diff
@@ -5590,7 +5590,7 @@ l = use_blocksize;
 						core->num->value = r_core_print_disasm (core->print,
 							core, addr, block1, addrbytes * l, l, 0, 1, formatted_json, NULL, NULL);
 					} else {
-						eprintf ("Cannot allocate %d byte(s)\n", addrbytes * l);
+						eprintf ("Cannot allocate %ld byte(s)\n", addrbytes * l);
```
Comment (radare) [inline libr/core/cmd_print.c:5593]: as i said, addrbytes is int in r_io, i would probably go for casting (int) instead of changing this format, the value of addrbytes shouldnt be larger than 8

File: libr/socket/run.c [hunk 8]
```diff
@@ -162,15 +162,16 @@ static char *getstr(const char *src) {
...
-				int i, len, rep;
+				size_t len;
+				long i, rep;
 				*pat++ = 0;
-				rep = atoi (src + 1);
+				rep = strtol (src + 1, NULL, 10);
 				len = strlen (pat);
 				if (rep > 0) {
-					char *buf = malloc (rep);
+					char *buf = malloc (rep * sizeof(char));
```
Comment (radare) [inline libr/socket/run.c:171]: missing space before (

File: libr/core/cmd_print.c (hunk 2 above)
Comment (08A) [reply libr/core/cmd_print.c]: swap back to the old type

**Comment by radare** [comment]
This pull request introduces 1 alert

**Comment by radare** [comment]
thats the introduced alert

Screenshot 2020-04-11 at 20 19 52

**Comment by radare** [comment]
this should be merged tomorrow or before, lets focus on fixing issues for the release now

thanks!

**Comment by 08A** [comment]
> 
> 
> lgtm except for the remaining issue . bear in mind that sizeof(char) is always 1, so the multiplication doesnt solves anything there

I replaced atoi with strtol, which is a better function.
I didn't use sizeof(char) to solve the issue, it was just for consistency.

**Comment by radare** [comment]
It breaks the windows build

**Comment by 08A** [comment]
The problem was the use of ssize_t.

**Comment by radare** [comment]
This pull request introduces 1 alert
